modules/mbedtls/zephyr_init.c

```diff
@@ -134,13 +141,17 @@ psa_status_t mbedtls_psa_external_get_random(
 	uint8_t *output, size_t output_size, size_t *output_length)
 {
 	(void) context;
+	int ret;
 
-	if (sys_csrand_get(output, output_size) != 0) {
+	ret = get_random_data(output, output_size,
+		IS_ENABLED(CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG) ?
+		false : true);
```

Comment:
Suggest simplifying this by removing the ternary operator and turning `cs_only` in `get_random_data()` into `allow_non_cs`:
```suggestion
		IS_ENABLED(CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG));
```